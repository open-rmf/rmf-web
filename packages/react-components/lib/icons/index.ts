export * from './CloseFullscreen';
export * from './OpenInFull';
