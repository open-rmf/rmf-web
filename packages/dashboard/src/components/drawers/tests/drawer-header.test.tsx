import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import DrawerHeader from '../drawer-header';
import CloseIcon from '@material-ui/icons/Close';
import { KeyboardBackspace as BackIcon } from '@material-ui/icons';
import { Typography } from '@material-ui/core';

const mount = createShallow();

describe('Drawer header', () => {
  test('Renders correctly', () => {
    const root = mount(
      <DrawerHeader
        handleCloseButton={() => jest.fn()}
        handleBackButton={() => jest.fn()}
        title={'Test'}
      />,
    );
    expect(root.find(Typography).html()).toContain('Test');
    expect(root.find(CloseIcon).exists()).toBeTruthy();
    expect(root.find(BackIcon).exists()).toBeTruthy();
    expect(root).toMatchSnapshot();
  });

  test('Renders without back button if there is no handler', () => {
    const root = mount(<DrawerHeader handleCloseButton={() => jest.fn()} title={'Test'} />);
    expect(root.find(BackIcon).exists()).toBeFalsy();
  });

  test('Click on close button fires an event correctly', () => {
    let clicked = false;
    const root = mount(<DrawerHeader handleCloseButton={() => (clicked = true)} title={'Test'} />);
    root.find('#closeDrawerButton').simulate('click');
    expect(clicked).toBe(true);
  });

  test('Click on back button fires an event correctly', () => {
    let clicked = false;
    const root = mount(
      <DrawerHeader
        handleCloseButton={() => jest.fn()}
        handleBackButton={() => (clicked = true)}
        title={'Test'}
      />,
    );
    root.find('#backDrawerButton').simulate('click');
    expect(clicked).toBe(true);
  });
});
