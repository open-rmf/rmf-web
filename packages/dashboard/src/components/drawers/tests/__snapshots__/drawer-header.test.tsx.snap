// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer header Renders correctly 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <WithStyles(ForwardRef(IconButton))
      className="makeStyles-button-2"
      id="backDrawerButton"
      onClick={[Function]}
    >
      <KeyboardBackspaceIcon />
    </WithStyles(ForwardRef(IconButton))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    alignItems="center"
    container={true}
  >
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-heading-1"
      item={true}
    >
      <WithStyles(ForwardRef(Typography))
        variant="h6"
      >
        Test
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <WithStyles(ForwardRef(IconButton))
        className="makeStyles-button-2"
        id="closeDrawerButton"
        onClick={[Function]}
      >
        <CloseIcon />
      </WithStyles(ForwardRef(IconButton))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Divider)) />
</Fragment>
`;
